### PR TITLE
Implement random library based on qrng.anu.edu.au

### LINKED ISSUE #265: Implement random library based on http://qrng.anu.edu.au/
https://qrng.anu.edu.au/API/jsonI.php?length=5&type=uint8 returns 5 random uint8 numbers

### PR DESCRIPTION
**Pull request by prin-r**
Fixes #265

### CODE REVIEW

File: owasm/src/ext/random/qrng_anu.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        ShellCmd::new(
+            "curl",
+            &[format!("https://qrng.anu.edu.au/API/jsonI.php?length={}&type=uint8", &self.size)],
+        )
+    }
+
+    fn from_cmd_output(&self, output: String) -> Option<Vec<u8>> {
+        let parsed = json::parse(&output).ok()?;
+        let bytes = parsed["data"].members().map(|x| x.as_u8()).collect::<Option<Vec<u8>>>()?;
+        if bytes.len() as u8 == self.size {
```
Comment (sorawit) [inline owasm/src/ext/random/qrng_anu.rs]: Will this pass if, for example, `bytes.len()` is 257 and `self.size` is 1? 🤔

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]

File: owasm/example/src/logic.rs [hunk 2]
```diff
@@ -12,28 +13,40 @@ decl_data! {
...
     }
 }
 
 impl Data {
     pub fn avg_px(&self) -> f32 {
         (self.coin_gecko + self.crypto_compare + self.binance) / 3.0
     }
+
+    pub fn rng_to_u64(&self) -> u64 {
+        let mut acc: u64 = 0;
```
Comment (taobun) [inline owasm/example/src/logic.rs]: Use parse vec<u8> to u64 instead sum

File: owasm/src/ext/random/qrng_anu.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        assert_eq!(
+            RandomBytes::new(10).from_cmd_output(r#"{"type":"uint8","length":10,"data":[29,21,227,184,32,128,97,128,231,197],"success":true}"#.into()),
+            Some(vec![29,21,227,184,32,128,97,128,231,197])
+        );
+    }
+
+    #[test]
+    fn test_from_cmd_not_ok() {
+        assert_eq!(RandomBytes::new(10).from_cmd_output(r#"{"success":false}"#.into()), None);
+    }
```
Comment (taobun) [inline owasm/src/ext/random/qrng_anu.rs:66]: I want another test in length of result must equal to size parameter

**Review by taobun — CHANGES_REQUESTED** [review]

File: owasm/example/src/logic.rs [hunk 4]
```diff
@@ -75,11 +91,17 @@ mod tests {
             binance: 210.0,
             alphavantage: 220.0,
             time_stamp: 12,
+            rng: vec![8, 7, 6, 5, 4, 3, 2, 1],
         };
-        // Average among the two data points is 170.00
+        // Average among the two data points is 175.00
```
Comment (sorawit) [inline owasm/example/src/logic.rs]: Why is this changed?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]
lgtm

**Review by taobun — APPROVED** [review]

**Pull request merged**